docker/service/src/main/java/uk/nhs/adaptors/scr/fhirmappings/ObservationMapper.java

```
@@ -0,0 +1,104 @@
+            }
+        }
+    }
+
+    private static void getObservationStatus(ObservationObject observationObject, Observation observation) {
+        if (observation.hasStatus()) {
+            Optional<String> status = getHL7Status(observation.getStatus().toString().toLowerCase());
+            status.ifPresent(observationObject::setObservationStatus);
+        }
+    }
+
+    private static void getObservationStartTime(ObservationObject observationObject, Observation observation) throws FhirMappingException {
+        if (observation.hasEffectiveDateTimeType()) {
+            String strDate = observation.getEffectiveDateTimeType().getValueAsString();
+            observationObject.setObservationStartTime(DateUtil.formatDateFhirToHl7(strDate));
+        }
+        if (observation.hasEffectivePeriod()) {
+            DateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss'Z'");
+            Period period = observation.getEffectivePeriod();
+            if (period.hasStart()) {
+                String strDate = dateFormat.format(period.getStart());
+                observationObject.setObservationStartTime(DateUtil.formatDate(strDate));
+            }
+            if (period.hasEnd()) {
+                String strDate = dateFormat.format(period.getEnd());
+                List<EndTime> endTimeList = new ArrayList<>();
+                EndTime endTime = new EndTime();
+                endTime.setEndTime(DateUtil.formatDate(strDate));
+                endTimeList.add(endTime);
+                observationObject.setObservationEndTimeList(endTimeList);
+            }
+        }
+    }
+
+    private static Optional<String> getHL7Status(String status) {
+        switch (status) {
+            case "entered-in-error":
+                return Optional.of("nullified");
+            case "final":
+                return Optional.of("completed");
+            default:
+                return Optional.empty();
```

I would throw an exception